app/uk/gov/hmrc/estates/services/DesService.scala

```diff
@@ -38,8 +43,33 @@ class DesService @Inject()(val desConnector: DesConnector) {
     desConnector.getSubscriptionId(trn)
   }
 
-  def getEstateInfo(utr: String): Future[GetEstateResponse] = {
-    desConnector.getEstateInfo(utr)
+  def refreshCacheAndGetTrustInfo(utr: String, internalId: String)(implicit hc: HeaderCarrier): Future[GetEstateResponse] = {
+    Logger.debug("Retrieving Estate Info from DES")
+    Logger.info(s"[DesService][refreshCacheAndGetTrustInfo] refreshing cache")
+
+    repository.resetCache(utr, internalId).flatMap { _ =>
+      desConnector.getEstateInfo(utr).map {
+        case response: GetEstateProcessedResponse =>
+          repository.set(utr, internalId, Json.toJson(response)(GetEstateProcessedResponse.mongoWrites))
```

Comment:
Why don't we wait for the future here? 🤔

Reply:
Fixed now